morituri/command/cd.py

```diff
@@ -134,10 +134,26 @@ def do(self):
         if not self.program.metadata:
             # fall back to FreeDB for lookup
             cddbid = self.ittoc.getCDDBValues()
-            cddbmd = self.program.getCDDB(cddbid)
-            if cddbmd:
-                sys.stdout.write('FreeDB identifies disc as %s\n' % cddbmd)
 
+            allow_freedb = getattr(self.options, 'allow_freedb', False)
+
+            sfdb = simplefreedb.SimpleFreeDB()
+            matches = sfdb.query(cddbid[0], cddbid[1], cddbid[2:-1], cddbid[-1])
+            if matches:
+                match = matches[0] # TODO: honor --prompt
+
+                sys.stdout.write('FreeDB identifies disc as %s\n' % match['artist_title'])
+
+                if allow_freedb:
+                    logger.warning('Using FreeDB metadata, which is not '
+                                   'recommended due to its very low quality '
+                                   'standards.')
+                    logger.warning('You are strongly encouraged to submit CD '
+                                   'information to the MusicBrainz database.')
+                    freedb_data = sfdb.read(match['category'], match['discid'])
+                    self.program.metadata = self.program.craftMusicBrainzFromFreeDB(freedb_data)
+
+        if not self.program.metadata:
```

Comment:
Following the convention of failing early:

```python
if matches:
    ...
    if not self.options.allow_freedb and not self.options.unknown:
        logger.critical(...)
        return -1
    logger.warning(...)
    freedb_data ...
    ...
```

Reply:
I'd like to keep the diff as small as possible, and I'm not sure I'm following what you are trying to restructure here. I'd rather have such change on top of the proposed change if you don't mind.